### DISCUSSION TITLE
Converting data to gluon-ts format

### DISCUSSION THREAD

**Discussion by sege01** [awslabs/gluonts#1582] [Q&A]
Hello,
Thank you for the excellent library.
I am new to gluon-ts and trying to learn. I have some queries.
I am trying to use stallion_data(https://www.kaggle.com/utathya/future-volume-prediction) for volume forecasting.

How can I convert this data to gluon-ts Listdatset format to further pass it to the model?
Data looks like this:- 
![image](https://user-images.githubusercontent.com/85530006/122917064-e60d9080-d35d-11eb-996a-afda01fc7c30.png)

In the dataframe, each entry consists of 'Agency' and 'SKU', whose combination makes an entry unique. 
So, according to that, the one-time series will be consisting of the unique combination of 'agency' and 'SKU' with data for a certain period(start date to end date).

Thank you in advance.

**Comment by lostella** [comment]
@sege01 to construct a `ListDataset` you need to do the following

```python
dataset = ListDataset(
    [
        {"start": <pandas Timestamp>, "target": <list of values>},
        {"start": <pandas Timestamp>, "target": <list of values>},
        ...
    ],
    freq="M"  # the data appears to be monthly?
)
```

where each dictionary in the list is a time series with start timestamp and values. I suggest you take a look at the [`groupby`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.groupby.html) operation in order to process the data frame.

**Reply by sege01** [reply]
Thank you so much @lostella for the reply.
I had one more question if you could provide some suggestions.

I converted the data in gluonts format but  I needed some help in understanding how one can handle missing data. For example, in the case of demand forecasting, there could be some products for which data is not available like snow boots are not available during summer. So, there is no data for snow boots for summer. How this kind of gap in data can be handled? 
The frequency of data is 'W' (weekly) which means for different products/time series data is missing for different time periods (weeks).

Thanking you in advance.

**Reply by mbohlkeschneider** [reply]
You can encode missing values by `NaN`. If your chosen model is able to handle missing data, it will do so automatically. If not, it will fail.
